### PR TITLE
Display standard output in the emacs plugin even when ocamlformat does not fail

### LINKED ISSUE #1170: Feature request: show parse-docstrings warnings in emacs
Currently, when there are warnings from `parse-docstrings = true`, they are not shown by the emacs integration. When invoking ocamlformat from a shell, the warnings are sent to stderr, and ocamlformat exits with success `0`. In this case, invoking ocamlformat using the emacs integration silently swallows the warnings. It would be nice to (have an option to) show these warnings.

### PR DESCRIPTION
**Pull request by gpetiot**
Fix #1170

### CODE REVIEW

File: emacs/ocamlformat.el [hunk 1]
```diff
@@ -261,14 +270,11 @@ function."
...
                (message "Applied ocamlformat"))
-             (if errbuf
-               (progn
-                 (with-current-buffer errbuf
-                   (setq buffer-read-only nil)
-                   (erase-buffer))
-                 (ocamlformat--process-errors
-                   (buffer-file-name) bufferfile errorfile errbuf)))
+             (ocamlformat--process-errors
+               (buffer-file-name) bufferfile errorfile errbuf)
```
Comment (Julow) [inline emacs/ocamlformat.el]: You could also call it from outside this if, after `message`.

File: emacs/ocamlformat.el [hunk 2]
```diff
@@ -157,17 +157,26 @@ function."
...
-      (insert "ocamlformat errors:\n")
-      (while (search-forward-regexp (regexp-quote tmpfile) nil t)
-        (replace-match (file-name-nondirectory filename)))
-      (compilation-mode)
-      (display-buffer errbuf))))
+  (if errbuf
+    (progn
+      (with-current-buffer errbuf
+        (setq buffer-read-only nil)
+        (erase-buffer))
```
Comment (Julow) [inline emacs/ocamlformat.el]: It was already like this but wouldn't make more sense to do that in the "not `'echo`" case ?

File: emacs/ocamlformat.el [hunk 3]
```diff
@@ -157,17 +157,26 @@ function."
...
+      (with-current-buffer errbuf
+        (if (eq ocamlformat-show-errors 'echo)
+          (message "%s" (buffer-string))
+          ;; Checks the size of errorfile
+          (if (> (nth 7 (file-attributes errorfile)) 0)
```
Comment (Julow) [inline emacs/ocamlformat.el]:
```suggestion
          (if (> (file-attribute-size (file-attributes errorfile)) 0)
```
This works for me :)

Comment (gpetiot) [reply emacs/ocamlformat.el]: I added a comment to keep track of why it's not the case

File: emacs/ocamlformat.el (hunk 2 above)
Comment (gpetiot) [reply emacs/ocamlformat.el]: done

File: emacs/ocamlformat.el (hunk 1 above)
Comment (gpetiot) [reply emacs/ocamlformat.el]: done

**Pull request merged**